trestle/core/commands/author/ssp.py

```diff
@@ -383,6 +383,23 @@ def _merge_comp_defs(
                     if new_ssp:
                         SSPAssemble._add_imp_req_to_ssp(ssp, gen_comp, imp_req, set_params, context)
                     else:
+                        # compile all new uuids for new component definitions
+                        comp_uuids = [x.uuid for x in comp_dict.values()]
+                        for imp_requirement in as_list(ssp.control_implementation.implemented_requirements):
+                            diffs = [
+                                x.component_uuid
+                                for x in imp_requirement.by_components
+                                if x.component_uuid not in comp_uuids
+                            ]
+                            for diff in diffs:
+                                logger.warning(
+                                    f'By_component {diff} removed from implemented requirement '
+                                    f'{imp_requirement.control_id} '
```

Comment:
please add the reason it was removed - i.e. because the corresponding component is not in the specified compdefs

Reply:
Done